examples/cvector-generator/mini-tests/test-vanilla-pca.cpp

```diff
@@ -0,0 +1,116 @@
+
+#include "common.h"
+#include "llama.h"
+#include "ggml.h"
+#include "../pca.hpp"
+
+#ifdef GGML_USE_CUDA
+#include "ggml-cuda.h"
+#endif
+
+#ifdef GGML_USE_METAL
+#include "ggml-metal.h"
+#endif
+
+#include <cstdio>
+#include <cstring>
+
+// Function to run PCA and print results
+static void run_pca_test(struct ggml_context *ctx, float *matrix, int rows, int cols) {
+    // struct ggml_tensor *input_tensor = create_tensor(ctx, matrix, rows, cols);
+    struct ggml_tensor *input_tensor = ggml_new_tensor_2d(ctx, GGML_TYPE_F32, rows, cols);
+    memcpy(input_tensor->data, matrix, rows * cols * sizeof(float));
```

Comment:
Instead of reusing the same context, do something like this:

```suggestion
static void run_pca_test(float * matrix, int rows, int cols) {
    struct ggml_init_params ctx_params {
        /*.mem_size   =*/ ggml_tensor_overhead(),
        /*.mem_buffer =*/ NULL,
        /*.no_alloc   =*/ false,
    };
    ggml_context_ptr ctx { ggml_init(ctx_params) };
    struct ggml_tensor * input_tensor = ggml_new_tensor_2d(ctx, GGML_TYPE_F32, rows, cols);
    ggml_backend_buffer_ptr buffer { ggml_alloc_ctx_tensors_from_buft(ctx, ggml_backend_cpu_buffer_type()) };
    ggml_backend_tensor_set(input_tensor, matrix, 0, rows * cols * sizeof(float));
```

Reply:
Sounds good! I'll add the suggestion thanks!